### PR TITLE
647 clean up dual simulations

### LINKED ISSUE #647: Clean up `DualSimulations`
See https://github.com/Roche/crmPack/blob/main/R/Simulations-class.R#L168 and corresponding class `DualSimulations`

To do:
- [x] Clean up similarly as the other classes above in the file
    - [x] Add for simpler navigation in RStudio the lines with `# ClassName ----` and `## class ----` and `## constructor ----`
	- [x] includes changing to `snake_case` including downstream updates where needed:
		- [x] `rhoEst` to `rho_est`
		- [x] `sigma2West` to `sigma2W_est`
		- [x] `fitBiomarker` to `fit_biomarker`
	- [x] documentation cleanup:
    	- [x] replace double `##` with `#`
    	- [x] remove `@keywords class` line
    	- [x] replace `\code{\linkS4class{ClassName}}` with ``[`ClassName`]``
    	- [x] Use ``@slot model (`type or class`)\cr`` for documenting the type or class of the slots
    	- [x] Use simple title `ClassName` followed by the lifecycle badge, followed by the description of the class
    	- [x] Add the `@aliases` line for the class definition
	- [x] Use the argument names for `setClass` and replace `representation()` with simple `c()`
	- [x] For `contains` just use simple string instead of list with a single string
	- [x] remove the `validObject` line
	- [x] user constructor:
		- [x] Move the example here, and for consistency rename the file to start with upper case letter, e.g. `Design` instead of `design`
		- [x] Use `@rdname ClassName-class` to put the constructor documentation in the same place as the class definition
		- [x] Use ``@param model (`type or class`)\cr see slot definition.`` for documenting the arguments
		- [x] Remove `@keywords classes` line
		- [x] Remove `@return` line for now 
- [x] Clean up the example code file
- [x] Add unit tests to https://github.com/Roche/crmPack/blob/main/tests/testthat/test-Simulations-class.R
    - [x] compare other tests in other files and use the same approach
    - [x] in particular use `# bla bla ----` etc. section dividers to help navigation in RStudio IDE

### PR DESCRIPTION
**Pull request by pengguanya**
# Pull Request

Fixes #647

@danielinteractive please review

### CODE REVIEW

**Comment by pengguanya** [comment]
@danielinteractive The pull request is pending due to some building errors in vignettes/examples.Rmd that are not directly related to my updates and may link to other's updates. I am off next week and will continue the week after.

File: R/Simulations-class.R [hunk 1]
```diff
@@ -179,32 +179,36 @@ Simulations <- function(fit,
...
+## class ----
+
+#' `DualSimulations`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This class captures the trial simulations from dual-endpoint model based
+#' designs. In comparison to the parent class [`Simulations`],
+#' it contains additional slots to capture the dose-biomarker fits, and the
+#' sigma2W and rho estimates.
```
Comment (danielinteractive) [inline R/Simulations-class.R:193]: I would put these in backticks to avoid spell check issues

File: R/Simulations-class.R [hunk 2]
```diff
@@ -179,32 +179,36 @@ Simulations <- function(fit,
...
 .DualSimulations <-
   setClass(
     Class = "DualSimulations",
-    representation(
-      rhoEst = "numeric",
-      sigma2West = "numeric",
-      fitBiomarker = "list"
+    slots = c(
+      rho_est = "numeric",
+      sigma2_west = "numeric",
```
Comment (danielinteractive) [inline R/Simulations-class.R:205]:
```suggestion
      sigma2w_est = "numeric",
```
And similar elsewhere

**Review by danielinteractive — COMMENTED** [review]
Thanks @pengguanya

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**